lib/cocoapods/installer/analyzer.rb

```diff
@@ -722,9 +726,10 @@ def generate_version_locking_dependencies(podfile_state)
         else
           deleted_and_changed = podfile_state.changed + podfile_state.deleted
           deleted_and_changed += pods_to_update[:pods] if update_mode == :selected
-          local_pod_names = @podfile_dependency_cache.podfile_dependencies.select(&:local?).map(&:root_name)
+          local_pod_names = podfile_dependencies.select(&:local?).map(&:root_name)
           pods_to_unlock = local_pod_names.to_set.delete_if do |pod_name|
-            sandbox.specification(pod_name).checksum == lockfile.checksum(pod_name)
+            next unless sandbox_specification = sandbox.specification(pod_name)
```

Comment:
does this fix something else?

Reply:
oh nm I think this can fix a crasher.

Reply:
yeah, this is the actual fix, the other change is just to allow testing it